### PR TITLE
DOC: More details for conda installs

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR:
* Fixes the installation instructions from the intel channel to also add conda-forge, which is required for some dependencies.
* Adds a visible word of warning that the packages from the intel channel are meant to interoperate with conda-forge but not with anaconda packages.
* Modifies the suggested commands to avoid pulling anything from the anaconda channel when it is added as a default in user configs.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

### CODE REVIEW

File: doc/sources/distributed-mode.rst [hunk 1]
```diff
@@ -33,17 +33,19 @@ ensure that the spmd_backend is built.
 .. important::
   SMPD mode requires the |mpi4py| package used at runtime to be compiled with the same MPI backend as the |sklearnex|. The PyPI and Conda distributions of |sklearnex| both use Intel's MPI as backend, and hence require an |mpi4py| also built with Intel's MPI - it can be easily installed from Intel's conda channel as follows::
     
-    conda install -c https://software.repos.intel.com/python/conda/ mpi4py
+    conda install -c https://software.repos.intel.com/python/conda/ -c conda-forge --override-channels mpi4py
```
Comment (napetrov) [inline doc/sources/distributed-mode.rst:36]: for mpi4py i would just point to conda-forge. there is no need to intel channel there actually - this is 3rd party package

File: doc/sources/distributed_daal4py.rst [hunk 2]
```diff
@@ -47,7 +47,7 @@ same algorithms to much larger problem sizes.
     PyPI and conda are both compiled with `Intel's MPI <https://www.intel.com/content/www/us/en/developer/tools/oneapi/mpi-library.html>`__
     as MPI backend (offered as Python package ``impi_rt`` in both PyPI and conda): ::
 
-        conda install -c https://software.repos.intel.com/python/conda/ impi_rt
+        conda install -c https://software.repos.intel.com/python/conda/ -c conda-forge --override-channels impi_rt
```
Comment (napetrov) [inline doc/sources/distributed_daal4py.rst:50]: mpi_rt - it also available in conda forge so i would suggest just pointing there

File: doc/sources/quick-start.rst [hunk 3]
```diff
@@ -239,9 +252,19 @@ To prevent version conflicts, we recommend installing `scikit-learn-intelex` int
...
+
+        conda create -n sklex -c conda-forge --override-channels scikit-learn-intelex
+        conda activate sklex
+
+      To install in an existing environment: ::
+
+        conda install -c conda-forge --override-channels scikit-learn-intelex
 
-        conda install -c conda-forge scikit-learn-intelex
+      .. warning::
```
Comment (napetrov) [inline doc/sources/quick-start.rst:264]: should we have this in case of direct specification of conda-forge?  i mean other channels might be incompatible as well but we are not listing those

File: doc/sources/distributed_daal4py.rst [hunk 4]
```diff
@@ -61,7 +61,11 @@ same algorithms to much larger problem sizes.
     Intel's MPI backend can be easily installed from Intel's conda channel (see docs for
     :ref:`SPMD mode <distributed>` for more details): ::
 
-        conda install -c https://software.repos.intel.com/python/conda/ mpi4py
+        conda install -c https://software.repos.intel.com/python/conda/ -c conda-forge --override-channels mpi4py
```
Comment (yuejiaointel) [inline doc/sources/distributed_daal4py.rst:64]: Is --override-channels needed only when we have anaconda set as default?

File: doc/sources/distributed-mode.rst (hunk 1 above)
Comment (david-cortes-intel) [reply doc/sources/distributed-mode.rst:36]: But it uses OMPI as backend, which is not binary compatible with IMPI/MPICH.

File: doc/sources/distributed_daal4py.rst (hunk 4 above)
Comment (david-cortes-intel) [reply doc/sources/distributed_daal4py.rst:64]: Yes, that's precisely what the argument is for.

File: doc/sources/distributed_daal4py.rst (hunk 2 above)
Comment (david-cortes-intel) [reply doc/sources/distributed_daal4py.rst:50]: Updated.

File: doc/sources/quick-start.rst (hunk 3 above)
Comment (david-cortes-intel) [reply doc/sources/quick-start.rst]: Removed.

File: INSTALL.md [hunk 5]
```diff
@@ -107,28 +102,21 @@ We recommend this installation for the users of Intel® Distribution for Python.
...
+  conda install -c https://software.repos.intel.com/python/conda/ -c conda-forge scikit-learn-intelex
   ```
 
-> **_NOTE:_** If you do not specify the version of Python, the latest one is downloaded. 
-
-- Install into your current environment:
+_Note: packages from the Intel channel are meant to be used together with dependencies from the **conda-forge** channel, and might not
+work correctly when used in an environment where packages from the `anaconda` default channel have been installed. It is
+advisable to use the [miniforge](https://github.com/conda-forge/miniforge) installer for `conda`/`mamba`, as it comes with
+`conda-forge` as the only default channel._
```
Comment (ethanglaser) [inline INSTALL.md:118]:
```suggestion
**Note:** packages from the Intel channel are meant to be used together with dependencies from the **conda-forge** channel, and might not
work correctly when used in an environment where packages from the `anaconda` default channel have been installed. It is
advisable to use the [miniforge](https://github.com/conda-forge/miniforge) installer for `conda`/`mamba`, as it comes with
`conda-forge` as the only default channel.
```
To align syntax with other notes on this page

File: doc/sources/distributed-mode.rst [hunk 6]
```diff
@@ -33,18 +33,18 @@ ensure that the spmd_backend is built.
...
 
-  It also requires the MPI runtime executable (``mpiexec`` / ``mpirun``) to be from the same library that was used to compile the |sklearnex| - Intel's MPI runtime library is offered as a Python package ``impi_rt`` and will be installed together with the ``mpi4py`` package if executing the command above, but otherwise, it can be installed separately from different distribution channels:
-
-  - Intel's conda channel (recommended)::
+  .. warning:: Packages from the Intel channel are meant to be compatible with dependencies from ``conda-forge``, and might not work correctly in environments that have packages installed from the ``anaconda`` channel.
 
-      conda install -c https://software.repos.intel.com/python/conda/ impi_rt
+  It also requires the MPI runtime executable (``mpiexec`` / ``mpirun``) to be from the same library that was used to compile the |sklearnex| - Intel's MPI runtime library is offered as a Python package ``impi_rt`` and will be installed together with the ``mpi4py`` package if executing the command above, but otherwise, it can be installed separately from different distribution channels:
```
Comment (ethanglaser) [inline doc/sources/distributed-mode.rst:40]:
```suggestion
  It also requires the MPI runtime executable (``mpiexec`` / ``mpirun``) to be from the same library that was used to compile |sklearnex|. Intel's MPI runtime library is offered as a Python package ``impi_rt`` and will be installed together with the ``mpi4py`` package if executing the command above, but otherwise, it can be installed separately from different distribution channels:
```

File: doc/sources/distributed-mode.rst [hunk 7]
```diff
@@ -33,18 +33,18 @@ ensure that the spmd_backend is built.
...
 
   - Conda-Forge::
 
       conda install -c conda-forge impi_rt
 
+  .. tip:: ``impi_rt`` is also available from the intel channel: ``https://software.repos.intel.com/python/conda``.
```
Comment (ethanglaser) [inline doc/sources/distributed-mode.rst:46]:
```suggestion
  .. tip:: ``impi_rt`` is also available from the Intel channel: ``https://software.repos.intel.com/python/conda``.
```

**Review by ethanglaser — APPROVED** [review]
LGTM
